semver/semver.go

```diff
@@ -0,0 +1,117 @@
+package semver
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+const (
+	RegexSemver     = `^(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(-([0-9A-Za-z-]+(\.[0-9A-Za-z-]+)*))?(\+([0-9A-Za-z-]+(\.[0-9A-Za-z-]+)*))?$`
+	RegexPreRelease = `^(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)-([0-9A-Za-z-]+(\.[0-9A-Za-z-]+)*)`
+)
+
+type SemVer struct {
+	major      int
+	minor      int
+	patch      int
+	preRelease string
+	build      string
+}
+
+func ParseSemver(input string) (SemVer, error) {
```

Comment:
Change this to Parse as the package name will already be used while invoking . semver.Parse(<>)